libscript/sources/Engine.cxx

```diff
@@ -0,0 +1,73 @@
+#include "pivot/script/Engine.hxx"
+#include "Logger.hpp"
+#include "pivot/script/Exceptions.hxx"
+
+namespace pivot::ecs::script
+{
+
+Engine::Engine(systems::Index &systemIndex, component::Index &componentIndex)
+    : _systemIndex(systemIndex), _componentIndex(componentIndex)
+{
+}
+
+std::string Engine::loadFile(const std::string &file, bool isContent, bool verbose)
+{    // Load a file to register all descriptions in the file, as well as store the trees for the system entry points
+    if (verbose) std::cout << "Loading file " << file << std::endl;
+    // TODO: return string with error on top of the node
+    Node fileNode = parser::ast_from_file(file, isContent, verbose);    // generate abstract syntax tree from file
+    if (verbose) parser::printFileNode(fileNode);
+
+    // TODO: return string with error on top of the node
+    std::vector<systems::Description> sysDescs = interpreter::registerDeclarations(
+        fileNode, _componentIndex);    // register component descriptions, and retrieve system descriptions
+    for (systems::Description &desc: sysDescs) {
+        desc.system = std::bind(&Engine::systemCallback, this, std::placeholders::_1, std::placeholders::_2,
+                                std::placeholders::_3);    // add the callback to the descriptions
+        _systemIndex.registerSystem(desc);                 // register system in the not-so-global index
+        try {
+            Node systemEntry = getEntryPointFor(
+                desc.name, fileNode);    // only store the entry point for the system, the rest is in the description
+            _systems.insert({desc.name, systemEntry});
+        } catch (std::exception e) {
+            return e.what();    // shouldn't happen
+        }
+    }
+    return (file + " parsed succesfully.");    // no errors
+}
+
+void Engine::systemCallback(const systems::Description &system, component::ArrayCombination &entities,
+                            const event::EventWithComponent &trigger)
+{
+    if (!_systems.contains(system.name)) {
+        // std::cerr << std::format("Unregistered system '{}'", system.name) << std::endl; // format not available in
+        // c++20 gcc yet
+        logger.err("Unregistered system '") << system.name << "'";
+        return;
+    }
+    const Node &systemEntry = _systems.at(system.name);    // Avoid looking up for every entity
+    for (auto entity: entities) {                          // For every entity, execute the system with it as parameter
+        try {
+            _stack.clear();
+            interpreter::executeSystem(systemEntry, system, entity, trigger, _stack);
+        } catch (InvalidException e) {
+            logger.warn(e.getExceptionType()) << " '" << e.getFaulter() << "' => " << e.getError();
```

Comment:
I see why you want to have multiple char * value.
For now, you can use the `std::runtime_error` technique I mentioned earlier, but take a lot at the file "PivotException.hxx" in the lib graphics. It is more or less the same needs